backend/models/Word_Freq.js

```diff
@@ -0,0 +1,51 @@
+import natural from 'natural'
+const { WordTokenizer, PorterStemmer } = natural
+
+export function cleanAndTokenize(text) {
+    const stopWords = new Set([" ","m","i", "me", "my", "myself", "we", "our", "ours", "ourselves", "you", "your", "yours", "yourself", "yourselves", "he", "him", "his", "himself", "she", "her", "hers", "herself", "it", "its", "itself", "they", "them", "their", "theirs", "themselves", "what", "which", "who", "whom", "this", "that", "these", "those", "am", "is", "are", "was", "were", "be", "been", "being", "have", "has", "had", "having", "do", "does", "did", "doing", "a", "an", "the", "and", "but", "if", "or", "because", "as", "until", "while", "of", "at", "by", "for", "with", "about", "against", "between", "into", "through", "during", "before", "after", "above", "below", "to", "from", "up", "down", "in", "out", "on", "off", "over", "under", "again", "further", "then", "once", "here", "there", "when", "where", "why", "how", "all", "any", "both", "each", "few", "more", "most", "other", "some", "such", "no", "nor", "not", "only", "own", "same", "so", "than", "too", "very", "s", "t", "can", "will", "just", "don", "should", "now"])
+
+    const tokenizer = new WordTokenizer()
+    const tokens = tokenizer.tokenize(text.toLowerCase().replace(/[^\w\s]/g, ' '))
+    
+    const removeStopWords = tokens.filter(word => !stopWords.has(word))
+    return removeStopWords
+}
+
+export function cleanAndTokenizeNGrams(text) {
+    const spaces = new Set([' ', ""," ", ''])
+    const lower = text.toLowerCase()
+    //replace(/[^\w\s]/g, '') before split
+    const words = lower.split(/\s+/).filter(word => !spaces.has(word))
+    return words
+}
+
+export function generateNGrams(tokens, n) {
+    const nGrams = []
+    for (let i = 0; i <= tokens.length - n; i++) {
+        nGrams.push(tokens.slice(i, i + n).join(' '))
+    }
+    return nGrams
+}
+
+export function countFreq(words) {
+    const freq = {}
+    words.forEach(word => {
+        freq[word] = (freq[word] || 0) + 1
+    })
+    return freq
+}
+
+export function getFreq(text, n = 10, nGramSize = 1) {
+    const words = nGramSize===1 ? cleanAndTokenize(text) : cleanAndTokenizeNGrams(text)
```

Comment:
I've seen inconsistent formatting in the last PRs.
Were you able to install an auto-formatter? (early feedback I gave in the program)

Reply:
I went through all of the files to check and you are correct. The auto-formatter hasn't been running; the extension needed an update on VSCode. I have been using prettier as the auto formatter but it seems it hasn't been applying in recent files due to needing an update. That said, the issue has been resolved and prettier is now back up and running correctly. All current and future code will be formatted accordingly.